### PR TITLE
add server testing in node 0.10, 0.12 and iojs

### LINKED ISSUE #2700: CI testing of the server build
The server build should be added to the build matrix:

- iojs
- node 0.10 and 0.12

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8426242-ci-testing-of-the-server-build?utm_campaign=plugin&utm_content=tracker%2F22067&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F22067&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SergioCrisostomo**
CI testing of the server build (closes #2700).

Had to add some `xit : it` in the specs because they will now run in `node` environment and `document` and `window` are not there.

Added also some Travis optimizing and Safari 8 testing for Browser specs.

To run server specs locally we can use `grunt server`. 

Tested all this in Travis [and got green](https://travis-ci.org/SergioCrisostomo/mootools-core/builds/74641272) for Browser and node tests.

### CODE REVIEW

File: Specs/Core/Core.js [hunk 1]
```diff
@@ -610,7 +610,7 @@ describe('instanceOf', function(){
 	});
 
 	// todo(ibolmo)
-	if (window.Element && Element.set) it("should return true for Element instances", function(){
+	if (typeof window != 'undefined' && window.Element && Element.set) it("should return true for Element instances", function(){
```
Comment (arian) [inline Specs/Core/Core.js]: do the `typeof window` check in the `dit` definition?

File: Tests/gruntfile-options.js [hunk 2]
```diff
@@ -21,6 +26,14 @@ var packagerOptions = {
         src: ymlPackage.sources,
         dest: 'mootools-nocompat.js'
     },
+	server: {
```
Comment (arian) [inline Tests/gruntfile-options.js]: indentation.

File: package.json [hunk 3]
```diff
@@ -30,24 +30,25 @@
...
+    "karma-jasmine": "~0.1.5",
+    "karma-phantomjs-launcher": "~0.1.2",
+    "karma-requirejs": "~0.2.1",
     "karma-safari-launcher": "~0.1",
-    "js-yaml": "^3.0.2"
+    "karma-sauce-launcher": "~0.2.5",
+    "karma-script-launcher": "~0.1.0",
+    "karma-sinon": "~1.0.2",
+    "load-grunt-tasks": "~0.4.0",
+    "requirejs": "~2.1.11"
```
Comment (arian) [inline package.json]: Why do we need this?

File: Tests/gruntfile-options.js [hunk 4]
```diff
@@ -6,6 +6,11 @@ var ymlPackage = YAML.safeLoad(fs.readFileSync('package.yml', 'utf8'));
 var travisBuild = process.env.BUILD;
 var travisBrowser = process.env.BROWSER;
 var sauceBrowsers = JSON.parse(fs.readFileSync('Tests/browsers.json'));
+var serverBuild = {
+	modules: ['Core/Core', 'Core/Array', 'Core/String', 'Core/Number', 'Core/Function', 'Core/Object', 'Core/Class', 'Core/Class.Extras', 'Core/JSON'],
+	specFiles: ['Specs/Core/*.js', 'Specs/Core/*.js', 'Specs/Core/*.js', 'Specs/Core/*.js', 'Specs/Core/*.js'],
+	strip: ['1.2compat', '1.3compat', '1.4compat', '*compat', 'IE', 'ltIE8', 'ltIE9', '!ES5', '!ES5-bind'/*, '!ES6'*/, 'webkit', 'ltFF4']
```
Comment (arian) [inline Tests/gruntfile-options.js]: You just defined this in Tests/dist-tasks.json as well...

**Comment by SergioCrisostomo** [comment]
@arian thanks for reviewing. 

Updated. Removed  `requirejs` dependency that was there since old days.

**Comment by SergioCrisostomo** [comment]
@arian (or anyone else) can we merge this?

File: Specs/Core/Core.js [hunk 5]
```diff
@@ -967,25 +968,26 @@ describe('String.uniqueID', function(){
...
 
-	it("should return 'element' for an Element", function(){
+	dit("should return 'element' for an Element", function(){
 		var div = document.createElement('div');
 		expect(typeOf(div)).toEqual('element');
 	});
 
 	// todo(ibolmo)
-	if (window.Elements) it("should return 'elements' for Elements", function(){
+	if (typeof window != 'undefined' && window.Elements) dit("should return 'elements' for Elements", function(){
```
Comment (arian) [inline Specs/Core/Core.js]: isn't this `if` included in the `dit` function?

File: Gruntfile.js [hunk 6]
```diff
@@ -51,11 +54,21 @@ module.exports = function(grunt) {
...
 	var nocompatBuild = ['clean:specs', 'packager:nocompat', 'packager:specs-nocompat'];
-	var tasks = options.travis.build == 'default' ? compatBuild : nocompatBuild;
-	tasks = usePhantom ? tasks.concat('karma:continuous') : tasks.concat('karma:sauceTask');
+	var serverBuild = ['clean:specs', 'packager:server', 'packager:specs-server'];
+
+	var tasks = options.travis.build == 'default' ? compatBuild : options.travis.build == 'server' ? serverBuild : nocompatBuild;
+	tasks = options.travis.build == 'server' ? tasks.concat('server-jasmine-runner') : usePhantom ? tasks.concat('karma:continuous') : tasks.concat('karma:sauceTask');
+
+	// to run only server specs and avoid running browser specs for in the different versions of node and iojs
+	if (options.travis.build && options.travis.build != 'server' && process.version.slice(1, 5) != '0.10') tasks = [];
```
Comment (arian) [inline Gruntfile.js]: :+1:

**Pull request merged**